frontend/packages/kubevirt-plugin/src/utils/validations/template/template-validations.ts

```diff
@@ -33,6 +34,41 @@ export class TemplateValidations {
     return this.validateMemoryByType(value, ValidationErrorType.Warn);
   };
 
+  getAllowedBusses = (
+    validationErrorType: ValidationErrorType = ValidationErrorType.Error,
+  ): Set<DiskBus> => {
+    const allowedBusses = this.getAllowedEnumValues(
+      ValidationJSONPath.BUS,
+      validationErrorType,
+    ).map(DiskBus.fromString);
+
+    return new Set(allowedBusses.length === 0 ? DiskBus.getAll() : allowedBusses);
+  };
+
+  getRecommendedBusses = (): Set<DiskBus> => {
```

Comment:
Maybe we could implement it this way:
```ts
  getRecommendedBusses = (): Set<DiskBus> => {
    const allowedBuses = this.getAllowedBusses(ValidationErrorType.Error);
    const recommendedBuses = [...this.getAllowedBusses(ValidationErrorType.Warn)].filter((b) =>
      allowedBuses.has(b),
    );
    return new Set(recommendedBuses.length === 0 ? DiskBus.getAll() : recommendedBuses);
  };
```

this would make the errors more sensible as I was proposing in  https://github.com/openshift/console/pull/4105#discussion_r374305882

Reply:
OK. but the `return` statement should be in this case:
```javascript 
return new Set(recommendedBuses.length === 0 ? [...allowedBuses] : recommendedBuses);
```

Reply:
+1